src/monitor.rs

```diff
@@ -44,6 +47,51 @@ impl Iterator for AvailableMonitorsIter {
     }
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct VideoMode {
+    pub(crate) dimensions: (u32, u32),
+    pub(crate) bit_depth: u16,
+    pub(crate) frequency: u16,
+}
+
+impl VideoMode {
+    pub fn dimensions(&self) -> PhysicalSize {
+        self.dimensions.into()
+    }
+
+    pub fn bit_depth(&self) -> u16 {
+        self.bit_depth
+    }
+
+    pub fn frequency(&self) -> u16 {
+        self.frequency
+    }
+}
+
+/// An iterator over all available video modes.
+///
+/// Can be acquired with:
+/// - [`MonitorHandle::video_modes`][monitor_get].
+///
+/// [monitor_get]: ../monitor/struct.MonitorHandle.html#method.video_modes
+pub struct VideoModesIter<'a> {
```

Comment:
`type VideoModesIter<'a> = HashSetIter<'a, VideoMode>;`? No need to define our own struct and everything, I think.

Reply:
I think we ought to newtype it to avoid a breaking change if this type changes in the future, as the underlying collection, and hence the iterator type is an implementation detail. That is essentially what we're doing here.

Reply:
Would an type alias cause breaking changes if the details get changed? I wouldn't think so, but I haven't put much thought into the matter.

Reply:
I think it could, because it essentially acts the same as if the alias and the aliased type were equivalent.

As an example, let's say there's a function in user code that takes a `std::collections::hash_set::Iter` parameter (for some reason) instead of a generic `Iterator` parameter. If the alias in winit were suddenly to point to a `std::collections::vec_deque::Iter` instead, this would become a type check error. This is probably not how you would usually write a function that takes an iterator parameter, but it illustrates the problem.

I think different iterators do also expose different methods on them, so if some methods were suddenly to disappear, or new ones to appear entirely, this would be more visible of a breaking change.